public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -104,6 +106,12 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
       headers: {},
     });
 
+    for (const [key, value] of this.customParameters) {
+      if (data[key] == null) {
```

Comment:
`data` can be null/undefined.

Reply:
moved in query functions. So data is always defined.